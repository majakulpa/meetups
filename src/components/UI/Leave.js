import React from 'react'
import { HiOutlineTrash } from 'react-icons/hi'

const leave = ({
  question,
  textOne,
  textTwo,
  textThree,
  click,
  handleDelete,
  confirm
}) => {
  return (
    <div className="rounded border-solid border border-gray-200 bg-white px-5 py-8">
      <div className="text-center">
        <p className="text-lg">{question}</p>
        <span className="font-medium">
          '{textOne} {textTwo} {textThree}'
        </span>
      </div>
      <div className="flex justify-end mt-5">
        <button
          className="block bg-purple-600 float-right hover:bg-purple-800 text-white tracking-wide
       capitalize py-2 px-6 rounded focus:bg-purple-800 focus:outline-none focus:shadow-outline mr-3"
          onClick={click}
        >
          No, go back
        </button>
        <button
          className="block bg-gray-500 float-right hover:bg-gray-600 text-white tracking-wide flex
            capitalize py-2 px-4 rounded focus:bg-gray-800 focus:outline-none focus:shadow-outline"
          onClick={handleDelete}
        >
          <HiOutlineTrash className="mt-1 mr-1" />
          <span>{confirm}</span>
        </button>
      </div>
    </div>
  )
}

export default leave
