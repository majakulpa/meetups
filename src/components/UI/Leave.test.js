import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom/extend-expect'
import { cleanup } from '@testing-library/react'
import Leave from './Leave'
import renderer from 'react-test-renderer'

afterEach(cleanup)

test('if renders', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Leave />, div)
})

test('if matches snapshot', () => {
  const tree = renderer.create(<Leave />).toJSON()
  expect(tree).toMatchSnapshot()
})
