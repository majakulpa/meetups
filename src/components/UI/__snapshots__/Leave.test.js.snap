// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<div
  className="rounded border-solid border border-gray-200 bg-white px-5 py-8"
>
  <div
    className="text-center"
  >
    <p
      className="text-lg"
    />
    <span
      className="font-medium"
    >
      '
       
       
      '
    </span>
  </div>
  <div
    className="flex justify-center mt-5"
  >
    <button
      className="block bg-purple-600 float-right hover:bg-purple-800 text-white tracking-wide capitalize py-2 px-6 rounded focus:bg-purple-800 focus:outline-none focus:shadow-outline mr-3 text-xs sm:text-base"
      onClick={undefined}
    >
      No, go back
    </button>
    <button
      className="block bg-gray-500 float-right hover:bg-gray-600 text-white tracking-wide flex capitalize py-2 px-4 rounded focus:bg-gray-800 focus:outline-none focus:shadow-outline"
      onClick={undefined}
    >
      <svg
        className="hidden sm:block mt-1 mr-1"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
      <span
        className="text-xs sm:text-base"
      />
    </button>
  </div>
</div>
`;
